### PR TITLE
Prometheus: Code editor - upgrade /series API endpoints to use label/values and /labels for supported prometheus clients

### LINKED ISSUE #58187: Prometheus: Code editor - replace /series endpoint
Investigate removing current /series calls and migrating to more efficient API endpoints. This will probably require a small change in the functionality to query the label values separately from the labels themselves. 

See https://github.com/grafana/grafana/pull/58087 which implemented this change for the query builder.

### PR DESCRIPTION
**Pull request by gtk-grafana**
Replace usage of /series calls in prometheus code editor.

**What is this feature?**

Replace /series calls with labels and label values APIs, while maintaining backwards compatibility for incompatible prometheus versions.

**Why do we need this feature?**
Makes the code editor faster and uses less memory, especially for users that are querying high cardinality prometheus instances.

**Who is this feature for?**
Users of prometheus promQL editor, especially those with high cardinality prometheus instances.

**Which issue(s) does this PR fix?**:
Fixes https://github.com/grafana/grafana/issues/58187

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/language_provider.ts [hunk 1]
```diff
@@ -501,21 +502,60 @@ export default class PromQlLanguageProvider extends LanguageProvider {
...
     const urlParams = {
       ...range,
-      ...(interpolatedName && { 'match[]': match }),
+      ...(match && { 'match[]': match }),
     };
     return await this.request(`/api/v1/label/${interpolatedName}/values`, [], urlParams);
   };
 
+  /**
+   * Gets series labels
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/language_provider.ts:536]: Provide more context as to why there are two functions needed to replace a single one

File: public/app/plugins/datasource/prometheus/components/monaco-query-field/monaco-completion-provider/completions.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { escapeLabelValueInExactSelector } from '../../../language_utils';
 import { FUNCTIONS } from '../../../promql';
 
-import type { Situation, Label } from './situation';
+import type { Label, Situation } from './situation';
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/components/monaco-query-field/monaco-completion-provider/completions.ts:4]: Uhh, please revert?

**Review by gtk-grafana — COMMENTED** [review]
Not your worst PR, but it needs unit test coverage

File: public/app/plugins/datasource/prometheus/language_provider.ts [hunk 3]
```diff
@@ -501,19 +503,74 @@ export default class PromQlLanguageProvider extends LanguageProvider {
...
-    return await this.request(`/api/v1/label/${interpolatedName}/values`, [], urlParams);
+
+    const cacheParams = new URLSearchParams({
+      'match[]': interpolatedName ?? '',
+      start: roundSecToMin(parseInt(range.start, 10)).toString(),
+      end: roundSecToMin(parseInt(range.end, 10)).toString(),
+      name: name,
+    });
+
+    const cacheKey = `/api/v1/label/?${cacheParams.toString()}/values`;
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/language_provider.ts:541]: This is a bit of a regression risk, and will introduce cacheing of this query (rounding up to nearest minute) for both query builder and code editor.

**Review by itsmylife — APPROVED** [review]
LGTM

**Pull request merged**